src/components/Grid/snapping.js

```diff
@@ -10,7 +10,8 @@ export function gridSnapTargets(gridSpacing, { x, y }) {
   ].map(t => ({ ...t, type: 'grid' }));
 }
 
-function vertexSnapTargets(vertices, gridSpacing, location) {
+export function vertexSnapTargets(vertices, gridSpacing, location) {
+  if (vertices.length === 0) return [];
```

Comment:
This seems like a case that's already handled, yes? If we map over an empty list, we end up returning an empty list anyway